sonar-html-plugin/src/test/java/org/sonar/plugins/html/lex/PageLexerTest.java

```diff
@@ -503,8 +502,32 @@ public void cdata() {
   }
 
   @Test
-  public void tag_with_whitespace_before_name() {
-    assertSingleTag("<  html>");
+  public void tag_with_whitespace_after_name() {
+    assertSingleTag("<html  >");
+  }
+
+  @Test
+  public void tag_with_whitespace_before_name_is_considered_as_text() {
+    // Tag names cannot start with whitespace character: https://www.w3.org/TR/REC-xml/#sec-starttags.
+    assertOnlyText("<  html>");
+  }
+
+  @Test
+  public void start_tag_character_is_considered_as_text_when_followed_by_whitespace() {
+    PageLexer lexer = new PageLexer();
+    List<Node> nodes = lexer.parse(new StringReader("<a> < a </a>"));
+    assertThat(nodes).hasSize(3);
+    assertThat(nodes).extracting(Node::getNodeType).containsExactly(NodeType.TAG, NodeType.TEXT, NodeType.TAG);
+    assertThat(nodes).extracting(Node::getCode).containsExactly("<a>", " < a ", "</a>");
+  }
+
+  @Test
+  public void start_tag_character_is_considered_as_text_when_last_character_of_code() {
+    PageLexer lexer = new PageLexer();
+    List<Node> nodes = lexer.parse(new StringReader("<a> <"));
+    assertThat(nodes).hasSize(2);
+    assertThat(nodes).extracting(Node::getNodeType).containsExactly(NodeType.TAG, NodeType.TEXT);
+    assertThat(nodes).extracting(Node::getCode).containsExactly("<a>", " <");
```

Comment:
From my understanding, with this test, you are verifying that 
```java
return followingChar == 0 && ...
```
In the function, `isFollowedByWhitespaceCharacter` is working, right?